clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/CopyLastAsgAtomicOperation.groovy

```diff
@@ -275,10 +277,14 @@ class CopyLastAsgAtomicOperation implements AtomicOperation<DeploymentResult> {
         newDescription.ebsOptimized = description.ebsOptimized != null ? description.ebsOptimized : ebsOptimized
         newDescription.classicLinkVpcId = description.classicLinkVpcId != null ? description.classicLinkVpcId : classicLinkVPCId
         newDescription.classicLinkVpcSecurityGroups = description.classicLinkVpcSecurityGroups != null ? description.classicLinkVpcSecurityGroups : translateSecurityGroupIds(classicLinkVPCSecurityGroups)
-        newDescription.tags = description.tags != null ? description.tags : ancestorAsg.tags.collectEntries {
+        newDescription.tags = description.tags != null ? description.tags : ancestorAsg.tags?.collectEntries {
           [(it.getKey()): it.getValue()]
         }
-        newDescription.blockDevices
+//        newDescription.blockDevices  // todo(pdk27): FixMe. This seems like an incomplete LOC related to blockDevicesTags https://github.com/spinnaker/clouddriver/pull/5244/files.
```

Comment:
Is this changing existing behavior or just a note to self? Followed the PR but couldn't quite make the connection.

Reply:
Existing behavior as I understand this LOC is that block devices are not cloned. The LOC seems incomplete from the PR linked. 
So, this shouldn't change existing behavior, AFAIK. 

Follow-up question: Do you think blockDevices should be cloned? Wondering if they were left out in the original impl intentionally?  
IMO, the cloning of block devices should follow the exact logic [here](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/BasicAmazonDeployHandler.groovy#L587). Do you agree?

Reply:
I don't have the historical context on cloning/not cloning block devices. Not sure if the change it was introduced with requires this so I will defer.

But as far as general approach, I would think letting the logic you linked handle that is probably a good idea since it deals with changing instance type ect...